pal03377/Athena-New
# CoFee Adapter

This PR adds an adapter module for CoFee. Supported:

- sending submissions to CoFee
- getting back a response from CoFee (stored internally)
- sending feedback (only stored by the module itself)
- requesting feedback suggestions (based on the same logic which is currently in Artemis)

Not yet supported:
- submission selection by highest information gain

I'm also introducing a new feature to the playground where you can send all feedback on an exercise at once to a module for a better module testing experience.

# Steps for testing
- Start Athena CoFee (from https://github.com/ls1intum/Athena !)
- Start the Assessment Module Manager `cd assessment_module_manager && poetry install && poetry run assessment_module_manager`
- Start the CoFee Module `cd module_cofee && poetry install && poetry run module`
- Start the playground `cd playground && npm i && npm run dev`
- Visit the playground: http://localhost:3000. Select `module_cofee`
- Under "Send Submissions", select "3 text: Test Exercise with only 2 answers (but 10 submissions)", then send them
- Wait a bit and observe the logs
- Under "Send Feedback to Athena", select the same exercise and "All submissions". Send the feedback.
- Under "Request Feedback Suggestions from Athena", select the exercise again and any submission. Observe that you get at least one suggestion.

# Screenshots
![screenshot-2023-05-18_001361](https://github.com/pal03377/Athena-New/assets/9006596/05d71305-541b-4142-871b-6865e3765b59)
![screenshot-2023-05-18_001362](https://github.com/pal03377/Athena-New/assets/9006596/3e770312-a530-413d-a6cd-91db2e1e9dec)

![screenshot-2023-05-18_001360](https://github.com/pal03377/Athena-New/assets/9006596/616b1d59-21c9-48b0-8ef4-df1ab04e1e97)
🎉
